app/uk/gov/hmrc/helptosavereminder/repo/HtsReminderRepository.scala

```diff
@@ -176,24 +176,49 @@ class HtsReminderMongoRepository @Inject() (mongo: MongoComponent)(implicit val
       logger.warn(s"nextSendDate for User: ${htsReminder.nino} cannot be updated.")
       Future.successful(false)
     } else {
+      val modifiedJson = if (htsReminder.endDate.nonEmpty) {
+        List(
+          Updates.set("optInStatus", htsReminder.optInStatus),
+          Updates.set("email", htsReminder.email),
+          Updates.set("firstName", htsReminder.firstName),
+          Updates.set("lastName", htsReminder.lastName),
+          Updates.set("daysToReceive", htsReminder.daysToReceive),
+          Updates.set("endDate", htsReminder.endDate.get)
+        )
+      } else {
+        List(
+          Updates.set("optInStatus", htsReminder.optInStatus),
+          Updates.set("email", htsReminder.email),
+          Updates.set("firstName", htsReminder.firstName),
+          Updates.set("lastName", htsReminder.lastName),
+          Updates.set("daysToReceive", htsReminder.daysToReceive)
+        )
+      }
+
+      val updatedModifierJsonCallBackRef = if (htsReminder.callBackUrlRef.isEmpty) {
```

Comment:
Why all of this?  Why not something like htsReminder.callBackUrlRef.getOrElse("")

Reply:
you cant use getOrElse because it is not an option, the isEmpty is checking whether the string has length 0 or not